src/main/scala/scalaz/stream/io.scala

```diff
@@ -155,6 +155,33 @@ object io {
     } onHalt { _.asHalt }
 
   /**
+   * Create a Process from an iterator that requires some external or
+   * other mutable resource, while ensuring that the resource is released.
+   *
+   * Use `iterators` if the resource is associated with multiple iterators.
+   */
+  def iterator[R, O](acquire: Task[(R, Iterator[O])])(
+                     release: R => Task[Unit]): Process[Task, O] = {
+    bracket(acquire)(r => eval_(release(r._1))){
+      r => iteratorGo(r._2)
+    } onHalt { _.asHalt }
```

Comment:
I was basically copying `bracket` with a modification so that `release` is only called on the first value in the tuple returned by `acquire`. I don't know enough about the internals of scala-stream to say whether `asHalt` should be used here or in `bracket`.